src/compare.c

```diff
@@ -50,8 +50,9 @@ r_obj* vec_compare(r_obj* x, r_obj* y, bool na_equal) {
     case VCTRS_TYPE_integer:   COMPARE(int, r_int_cbegin, int_compare_na_equal);
     case VCTRS_TYPE_double:    COMPARE(double, r_dbl_cbegin, dbl_compare_na_equal);
     case VCTRS_TYPE_character: COMPARE(r_obj*, r_chr_cbegin, chr_compare_na_equal);
-    case VCTRS_TYPE_scalar:    r_abort("Can't compare scalars with `vec_compare()`");
-    case VCTRS_TYPE_list:      r_abort("Can't compare lists with `vec_compare()`");
+    case VCTRS_TYPE_complex:   r_abort("Can't compare complexes with `vec_compare()`.");
+    case VCTRS_TYPE_scalar:    r_abort("Can't compare scalars with `vec_compare()`.");
+    case VCTRS_TYPE_list:      r_abort("Can't compare lists with `vec_compare()`.");
```

Comment:
I think it's redundant to mention `vec_compare()` here, since it'll be part of the call?